### ISSUE TITLE
Addition is broken?

### ISSUE THREAD

**Issue by dpsanders** [JeffreySarnoff/FastRounding.jl#14]
On 0.6.4:

```
julia> FastRounding.add_round(0.1, 0.1, RoundUp)
0.2

julia> FastRounding.add_round(0.1, 0.1, RoundDown)
0.2
```

**Comment by dpsanders** [comment]
This seems to be an interaction between 0.6.4 and whichever versions I have installed?

**Comment by JeffreySarnoff** [comment]
if you use the newest tags (as yet not merged) of ErrorfreeArithmetic and FastRounding the issue should be resolved and working under both v0.6.4 and v0.7.0-beta2x.  Let me know.  
Idk why the FastRounding test on v0.6.4 was exiting with "ErrorfreeArithmetic not
 found in path" -- I'm guessing having made ErrorfreeArithmetic compatible with v0.6.4 will help -- meanwhile ignore that.

**Comment by JeffreySarnoff** [comment]
So, what was happening is that two_sum(0.1, 0.1) = (0.2, 0.0). While inexact, the inexactness of the sum is nonzero only beyond the second Float64 (the error).  As there is no way to tell whether the lo part (0.0) is above, below or at the true more-precise residual, I have made it so that RoundUp rounds 0.2 to nextfloat(0.2) and RoundDown rounds 0.2 to prevfloat(0.2).

**Comment by JeffreySarnoff** [comment]
I think I have things working together.  Use ErrorfreeArithmetic.jl ver0.1.2 (now merged) and reget FastRounding ver0.1.0 (tagged, not yet merged).

**Comment by dpsanders** [comment]
I don't think there should be any additional rounding done: the sum of the
floating point number 0.1 with itself is exactly the floating point number
0.2 (as two_sum is correctly telling us). This was my mistake.

**Comment by JeffreySarnoff** [comment]
except that 0.1 is not exactly representable, nor is 0.2:
```
julia> using DoubleFloats

julia> onetenth = Double64(1)/Double64(10)
9.9999999999999999999999999999999689e-02
julia> twotenths = onetenth + onetenth
1.9999999999999999999999999999999937e-01

julia> showtyped(onetenth)
Double64(0.1, -5.551115123125783e-18)
julia> showtyped(twotenths)
Double64(0.2, -1.1102230246251566e-17)
```
two_sum() --> (0.2, 0.0) is correct for two_sum, Double64 has me so its more accurate.

```
julia> setprecision(BigFloat,128)
128

julia> bf=BigFloat(2)/BigFloat(10)
2.000000000000000000000000000000000000001e-01

julia> Float64(bf),Float64(bf-Float64(bf))
(0.2, -1.1102230246251566e-17)
```

**Comment by JeffreySarnoff** [comment]
(FastRounding v0.1.0 is merged)

**Comment by dpsanders** [comment]
The additional rounding you added is breaking IntervalArithmetic tests.
Could you please remove it again? Sorry!

**Comment by dpsanders** [comment]
Neither the real number "0.1" nor "0.2" are representable.
But the exact sum of the floating-point number that is written "0.1" (i.e. the closest Float64 to the real number 1/10) with itself is exactly equal to the floating-point number written "0.2" (i.e. the closest Float64 to 2/10).

**Comment by JeffreySarnoff** [comment]
I can and I will but to really get true enclosures you need better than IEEE754 mandated rounding and the change does that.  I will add differently named options using the new logic and revert the current names.

**Comment by dpsanders** [comment]
Great, thanks.

I'm not sure what you mean by getting true enclosures -- isn't that what IntervalArithmetic.jl does?
Instead of Float64(0.1), you use `0.1..0.1`, which gives

```
julia> 0.1..0.1
Interval(0.09999999999999999, 0.1)
```
(after changing the output style with `@format full`).

**Comment by JeffreySarnoff** [comment]
tagged 0.1.1 awaiting merge
I am writing a different package with the new behavior -- by enclosure I
meant RoundUp/RoundDown would round to a value that, in all cases, was
above/below and as close as possible to the theoretical infinitely precise
result.  Doing that, unless I add checking for exact,  0.125+0..125 RoundUp/RoundDown is nextfloat(0.25)/prevfloat(0.25) .

**Comment by dpsanders** [comment]
Thanks!

The thing is that you don't know if by `0.1` the user meant "the real number 0.1" or "the floating-point number that is represented by the string '0.1'". We tried to go down this road in IntervalArithmetic, but in the end it's impossible to know what is meant, unless you use strings instead of floats.

**Comment by dpsanders** [comment]
Was fixed in 0.1.1.
